### PR TITLE
Allow --no-rbac flag that allows users to not pass rbac config

### PR DESCRIPTION
**Pull request by notfelineit**
Signed-off-by: notfelineit <notfelineit@gmail.com>

## Description
**Breaking Change**
This PR adds a flag `--no-rbac` that allows users to bypass passing the flag `--rbac-config`. In doing so, a default rbac configuration is used:
```
rules:
  - resource: "*"
    actions:
    - "get"
    - "create"
    - "delete"
    - "put"
    - "ping"
    subjects: ["*"]
    clusters: ["*"]
```

It also makes it so that a user must **explicitly** pass in either `--no-rbac` or `--rbac`, otherwise vtadmin will fail to come up:
Screen Shot 2022-03-23 at 4 59 44 PM

This is a **breaking change** since users who were using `--rbac-config` must now explicitly pass in `--rbac` flag as well.

## Cases
`--no-rbac`:
Screen Shot 2022-03-23 at 5 01 08 PM

`--rbac`:
Screen Shot 2022-03-23 at 4 59 24 PM

Failing to pass in either flag:
Screen Shot 2022-03-23 at 4 59 44 PM

## Related Issue(s)
This addresses this issue: https://github.com/vitessio/vitess/issues/9419

## Checklist
- [x] Should this PR be backported? **No**
- [ ] Tests were added or are not required
- [x] Documentation was added or is not required **Documentation should be added to release notes**

## Deployment Notes

N/A

### CODE REVIEW

File: go/cmd/vtadmin/main.go [hunk 1]
```diff
@@ -163,6 +171,8 @@ func main() {
 
 	// rbac flags
 	rootCmd.Flags().StringVar(&rbacConfigPath, "rbac-config", "rbac.yaml", "")
+	rootCmd.Flags().BoolVar(&enableRbac, "rbac", false, "whether to enable rbac")
+	rootCmd.Flags().BoolVar(&disableRbac, "no-rbac", false, "whether to disable rbac")
```
Comment (ajm188) [inline go/cmd/vtadmin/main.go:175]: i'm not sure which is better; i'm curious what you think about this!

```go
rootCmd.Flags().StringVar(&rbacConfigPath, "rbac-config", "", "path to an RBAC config file. must be set if not pasing --no-rbac") // now defaulting to empty
rootCmd.Flags().BoolVar(&disableRBAC, "no-rbac", false, "whether to disable RBAC. must be set if not passing --rbac-config")
```

and then in the run func:

```go
if rbacConfigPath == "" {
	if !disableRBAC {
		// fail, must pass exactly one
	}

	rbacConfig = rbac.DefaultConfig()
} else if disableRBAC {
	// fail, must pass exactly one
} else {
	cfg, err := rbac.LoadConfig(rbacConfigPath)
	if err != nil {
		fatal(err)
	}

	rbacConfig = cfg
}
```

File: go/vt/vtadmin/rbac/config.go [hunk 2]
```diff
@@ -163,3 +163,41 @@ func (c *Config) GetAuthenticator() Authenticator {
 func (c *Config) GetAuthorizer() *Authorizer {
 	return c.authorizer
 }
+
+func DefaultConfig() *Config {
+	log.Info("[rbac]: using default rbac configuration")
+	actions := []string{"get", "create", "delete", "put", "ping"}
```
Comment (ajm188) [inline go/vt/vtadmin/rbac/config.go:169]: we should use the exported constants here

File: go/vt/vtadmin/rbac/config.go [hunk 3]
```diff
@@ -163,3 +163,41 @@ func (c *Config) GetAuthenticator() Authenticator {
 func (c *Config) GetAuthorizer() *Authorizer {
 	return c.authorizer
 }
+
+func DefaultConfig() *Config {
```
Comment (ajm188) [inline go/vt/vtadmin/rbac/config.go:167]: this needs a doc comment

File: go/cmd/vtadmin/main.go [hunk 4]
```diff
@@ -43,6 +43,8 @@ var (
 	defaultClusterConfig cluster.Config
 
 	rbacConfigPath string
+	enableRbac     bool
+	disableRbac    bool
```
Comment (ajm188) [inline go/cmd/vtadmin/main.go:47]: per [Go guidelines](https://zchee.github.io/golang-wiki/CodeReviewComments/#initialisms), these should be written as "{enable,disable}RBAC" (fwiw, the reason `rbacConfigPath` is written that way is to not export it, and "rBAC" would be also incorrect)

**Review by ajm188 — COMMENTED** [review]
all i've got are style commentary!

File: go/cmd/vtadmin/main.go (hunk 1 above)
Comment (notfelineit) [reply go/cmd/vtadmin/main.go:175]: I like removing the default for the rbac config. I guess the other difference between this and the other is that the current implemention requires explicitly passing an "enable rbac" flag, specified by the issue: https://github.com/vitessio/vitess/issues/9419. I personally prefer explicitly enabling rbac per the issue 😄

**Review by ajm188 — COMMENTED** [review]
one more round 😅

File: go/cmd/vtadmin/main.go [hunk 5]
```diff
@@ -170,9 +170,9 @@ func main() {
 	rootCmd.Flags().BoolVar(&httpOpts.EnableDynamicClusters, "http-enable-dynamic-clusters", false, "whether to enable dynamic clusters that are set by request header cookies")
 
 	// rbac flags
-	rootCmd.Flags().StringVar(&rbacConfigPath, "rbac-config", "rbac.yaml", "")
-	rootCmd.Flags().BoolVar(&enableRbac, "rbac", false, "whether to enable rbac")
-	rootCmd.Flags().BoolVar(&disableRbac, "no-rbac", false, "whether to disable rbac")
+	rootCmd.Flags().StringVar(&rbacConfigPath, "rbac-config", "", "path to an RBAC config file. must be set if passing --rbac")
+	rootCmd.Flags().BoolVar(&enableRBAC, "rbac", false, "whether to enable rbac. must be set if not passing --rbac")
+	rootCmd.Flags().BoolVar(&disableRBAC, "no-rbac", false, "whether to disable rbac. must be set if not passing --no-rbac")
```
[the remark this answers is not in the document]
Comment (notfelineit) [reply go/cmd/vtadmin/main.go:175]: Whoo! Good idea

File: go/cmd/vtadmin/main.go [hunk 6]
```diff
@@ -103,19 +103,19 @@ func run(cmd *cobra.Command, args []string) {
...
 		if err != nil {
 			fatal(err)
 		}
 
 		rbacConfig = cfg
-	}
-	if disableRbac && !enableRbac {
-		rbacConfig = rbac.DefaultConfig()
+	} else if enableRBAC && rbacConfigPath == "" {
+		fatal("[rbac] must pass --rbac-config path when enabling rbac")
```
Comment (notfelineit) [inline go/cmd/vtadmin/main.go:116]:
```suggestion
		fatal("must pass --rbac-config path when enabling rbac")
```

File: go/cmd/vtadmin/main.go [hunk 7]
```diff
@@ -103,19 +103,19 @@ func run(cmd *cobra.Command, args []string) {
...
+	} else {
+		fatal("[rbac] must explicitly enable or disable RBAC by passing --no-rbac or --rbac")
```
Comment (notfelineit) [inline go/cmd/vtadmin/main.go:118]:
```suggestion
		fatal("must explicitly enable or disable RBAC by passing --no-rbac or --rbac")
```

**Pull request merged**